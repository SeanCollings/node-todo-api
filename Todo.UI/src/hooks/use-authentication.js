import { useDispatch, useSelector } from 'react-redux';
import { LOCAL_TOKEN } from '../constants';
import { LOGIN_USER_SUCCESS } from '../state/action-types';
import { decodeJWT, getCookie } from '../utils';

const useAuthentication = () => {
  const { user } = useSelector(({ auth }) => auth);
  const dispatch = useDispatch();

  if (!!user) return true;

  const token = getCookie(LOCAL_TOKEN);
  const username = decodeJWT(token);

  if (username?.name) {
    dispatch({
      type: LOGIN_USER_SUCCESS,
      payload: username,
    });
  }

  return !!username;
};

export default useAuthentication;
