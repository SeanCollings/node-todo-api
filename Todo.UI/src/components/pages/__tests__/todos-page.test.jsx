import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import TodosPage from '../todos-page';
import { useActions } from '../../../hooks/use-actions';

jest.mock('../../../hooks/use-actions');
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('Header', () => {
  const mockConfigState = {
    todos: { todos: [], error: null, loading: false },
    auth: { user: { name: 'john' } },
  };
  const mockConfigStateWithTodos = {
    todos: {
      todos: [
        { completed: false, task: 'First test task', id: 1 },
        { completed: true, task: 'Second test task', id: 2 },
      ],
      error: null,
      loading: false,
    },
    auth: { user: { name: 'john' } },
  };

  useActions.mockImplementation(() => ({
    addTodo: jest.fn(),
    getTodos: jest.fn(),
  }));

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render', () => {
    useSelector.mockReturnValueOnce(mockConfigState);

    const component = renderer.create(<TodosPage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render with todos', () => {
    useSelector.mockReturnValueOnce(mockConfigStateWithTodos);

    const component = renderer.create(<TodosPage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
