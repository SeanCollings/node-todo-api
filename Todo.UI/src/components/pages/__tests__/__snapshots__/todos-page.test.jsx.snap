// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div>
  <div
    className="sc-hKgILt ewOlPy"
  >
    Welcome, john
  </div>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    className="sc-bdfBwQ udtBZ"
    onClick={[Function]}
  >
    Add Todo
  </button>
</div>
`;

exports[`Header should render with todos 1`] = `
<div>
  <div
    className="sc-hKgILt ewOlPy"
  >
    Welcome, john
  </div>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    className="sc-bdfBwQ udtBZ"
    onClick={[Function]}
  >
    Add Todo
  </button>
  <div
    className="sc-fubCfw gYXWeR"
  >
    <div
      className="sc-eCssSg gldGQY"
    >
      <div>
        <input
          onChange={[Function]}
          value="First test task"
        />
      </div>
      <div>
        <input
          checked={false}
          className="sc-gKsewC dWswO"
          id={1}
          name={1}
          onChange={[Function]}
          type="checkbox"
          value={false}
        />
        <label
          className="sc-iBPRYJ jzDkHa"
          htmlFor={1}
        >
          Complete
        </label>
      </div>
      <div
        className="sc-jSgupP gjSJny"
      >
        <button
          className="sc-bdfBwQ sc-dlfnbm udtBZ gDwXjy"
          onClick={[Function]}
        >
          Delete
        </button>
        <button
          className="sc-bdfBwQ udtBZ"
          onClick={[Function]}
        >
          Update
        </button>
      </div>
    </div>
    <div
      className="sc-eCssSg fHIvyT"
    >
      <div>
        <input
          onChange={[Function]}
          value="Second test task"
        />
      </div>
      <div>
        <input
          checked={true}
          className="sc-gKsewC dWswO"
          id={2}
          name={2}
          onChange={[Function]}
          type="checkbox"
          value={true}
        />
        <label
          className="sc-iBPRYJ jzDkHa"
          htmlFor={2}
        >
          Complete
        </label>
      </div>
      <div
        className="sc-jSgupP gjSJny"
      >
        <button
          className="sc-bdfBwQ sc-dlfnbm udtBZ gDwXjy"
          onClick={[Function]}
        >
          Delete
        </button>
        <button
          className="sc-bdfBwQ udtBZ"
          onClick={[Function]}
        >
          Update
        </button>
      </div>
    </div>
  </div>
</div>
`;
